### PR TITLE
docs(ad-api): add fail safe api

### PR DESCRIPTION
**Pull request by isamu-takagi**
Signed-off-by: Takagi, Isamu <isamu.takagi@tier4.jp>

## Description

Add fail safe API.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The Reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: docs/design/autoware-interfaces/ad-api/list/index.md [hunk 1]
```diff
@@ -2,3 +2,4 @@
 
 - [Interface](./api/interface/index.md)
 - [Driving](./api/driving/index.md)
+- [Fail Safe](./api/fail_safe/index.md)
```
Comment (kenji-miyake) [inline docs/design/autoware-interfaces/ad-api/list/index.md:5]:
```suggestion
- [Fail-safe](./api/fail_safe/index.md)
```

File: docs/design/autoware-interfaces/ad-api/list/api/fail_safe/state.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+## Description
+
+Get the fail safe state. For details, see the [fail safe state](./index.md).
+
+## Message
+
+| Name          | Type   | Description                                         |
+| ------------- | ------ | --------------------------------------------------- |
+| state         | uint16 | A value of the [fail safe state](./index.md).       |
```
Comment (kenji-miyake) [inline docs/design/autoware-interfaces/ad-api/list/api/fail_safe/state.md:14]: [Question] How can we know the severity of the failure? I guess you'll add another message for that.

**Review by kenji-miyake — COMMENTED** [review]
LGTM, but added some small comments and questions.

File: docs/design/autoware-interfaces/ad-api/types/autoware_ad_api_msgs/msg/fail_safe_state.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+uint16 RTI_OPERATING = 2
+uint16 MRM_OPERATING = 3
+uint16 MRM_SUCCEEDED = 4
+uint16 MRM_FAILED = 5
+
+uint16 state
+bool auto_recovery
+```
+
+## This type uses
```
[the remark this answers is not in the document]
Comment (isamu-takagi) [reply docs/design/autoware-interfaces/ad-api/types/autoware_ad_api_msgs/msg/fail_safe_state.md:17]: For links to the types used by this type like [this](https://autowarefoundation.github.io/autoware-documentation/pr-155/design/autoware-interfaces/ad-api/types/autoware_ad_api_msgs/msg/route_optional/).

File: docs/design/autoware-interfaces/ad-api/list/api/fail_safe/state.md (hunk 2 above)
Comment (isamu-takagi) [reply docs/design/autoware-interfaces/ad-api/list/api/fail_safe/state.md:14]: I plan to add MRC such as stop, slow down and move to the shoulder. But first, let's assume that there is only one MRC.

**Review by kenji-miyake — APPROVED** [review]
LGTM!

**Pull request merged**